src/explorer/editors/FileEditor.ts

```diff
@@ -40,7 +41,9 @@ export class FileEditor extends BaseEditor<FileTreeItem> {
             throw new Error('Cannot update file after it has been closed.');
         }
         const localFile: Readable = fs.createReadStream(vscode.window.activeTextEditor.document.uri.fsPath);
-        node.etag = await putFile(node.root.client, localFile, node.path, nonNullProp(node, 'etag'));
+        let etag: string = nonNullValue(this._etags.get(node.fullId), 'etag');
+        etag = await putFile(node.root.client, localFile, node.path, etag);

```

Comment:
Is this for https://github.com/microsoft/vscode-azureappservice/issues/1209?

Reply:
I moved the etag stuff into this file because I don't need it in the shared package (everything is read only for functions). Plus it was always a little weird on the tree item (hence the [comment](https://github.com/microsoft/vscode-azureappservice/pull/1240/files#diff-9392209e5d5f170c97511a539a415a06L14) "cannot create etag on creation due to Kudu API calls"). The bug still repros after my changes